sats-dna/src/main/kotlin/com/sats/dna/components/UpcomingWorkoutListItem.kt

```diff
@@ -0,0 +1,241 @@
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import com.sats.dna.components.button.SatsButton
+import com.sats.dna.components.button.SatsButtonColor
+import com.sats.dna.components.icons.DefaultInstructorIcon
+import com.sats.dna.components.icons.InstructorType
+import com.sats.dna.components.icons.WorkoutType
+import com.sats.dna.components.icons.WorkoutTypeIcon
+import com.sats.dna.theme.SatsTheme
+import com.sats.dna.tooling.FontSizePreview
+import com.sats.dna.tooling.LightDarkPreview
+
+/**
+ * A section for a day of upcoming workouts
+
+ * @param day A formatted day that will be displayed as the title of the section.
+ * @param content The list of the containing workouts for this day. Consider using [UpcomingWorkoutListItem].
+ *
+ **/
+@Composable
+fun UpcomingWorkoutDaySection(
+    day: String,
+    content: @Composable ColumnScope.() -> Unit,
+) {
+    Column {
+        SatsSurface(Modifier.fillMaxWidth(), elevation = 1.dp) {
+            Text(
+                day,
+                Modifier.padding(
+                    vertical = SatsTheme.spacing.s,
+                    horizontal = SatsTheme.spacing.m,
+                ),
+            )
+        }
+        content()
+    }
+}
+
+/**
+ *  A list item for an upcoming workout.
+ *
+ * @param title The title of the workout.
+ * @param timeStamp A formatted date/time for when the workout started.
+ * @param location Where the workout took place, e.g. at which gym.
+ * @param duration The duration of the workout.
+ * @param onClick Callback to be invoked when this item is clicked.
+ * @param icon An icon to be displayed on the left gutter of the item.
+ * @param instructorInfo Information about the instructor for the workout.
+ * @param modifier The modifier to apply to the list item.
+ * @param unbookButton Optional slot for displaying an unbook button.
+ * @param friendsAttending Optional slot for displaying friends that are also attending the workout.
+ * **/
+@Composable
+fun UpcomingWorkoutListItem(
+    title: String,
+    timeStamp: String,
+    location: String,
+    duration: String,
+    icon: @Composable RowScope.() -> Unit,
+    instructorInfo: @Composable ColumnScope.() -> Unit,
+    modifier: Modifier = Modifier,
+    unbookButton: @Composable (RowScope.() -> Unit)? = null,
+    friendsAttending: @Composable (ColumnScope.() -> Unit)? = null,
+    onClick: () -> Unit,
+) {
+    Row(
+        modifier
+            .clickable(role = Role.Button) { onClick() }
+            .padding(SatsTheme.spacing.m),
+    ) {
+        Column(
+            verticalArrangement = Arrangement.spacedBy(SatsTheme.spacing.s),
+        ) {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.spacedBy(SatsTheme.spacing.m),
+                verticalAlignment = Alignment.CenterVertically,
```

Comment:
Vertically center-aligning these items looks a little bit strange to me. But that might just be me, though. :thinking: 

|   |   |
|---|---|
| ![image](https://github.com/sats-group/sats-dna-android/assets/386122/b72acf0d-6a1a-495d-b29b-847f38cf83bf) | ![image](https://github.com/sats-group/sats-dna-android/assets/386122/2bb01928-d906-4043-922f-175caa59f1e2) |

I'm on the fence. :sweat_smile:

Reply:
Let's rethink this with a designer later. :sunglasses:

Reply:
I realise that it's top aligned in the current version in prod, so I'll change it to that so it just stays the same mby? And then we can discuss if we should change it further down the line 🙏